adminSiteClient/ImagesIndexPage.tsx

```diff
@@ -0,0 +1,260 @@
+import React, {
+    useCallback,
+    useContext,
+    useEffect,
+    useMemo,
+    useState,
+} from "react"
+import { Button, Flex, Input, Space, Table, Upload } from "antd"
+
+import { AdminLayout } from "./AdminLayout.js"
+import { AdminAppContext } from "./AdminAppContext.js"
+import { DbEnrichedImage } from "@ourworldindata/types"
+import { Timeago } from "./Forms.js"
+import { ColumnsType } from "antd/es/table/InternalTable.js"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import { faUpload } from "@fortawesome/free-solid-svg-icons"
+import { Admin } from "./Admin.js"
+import { RcFile } from "antd/es/upload/interface.js"
+import TextArea from "antd/es/input/TextArea.js"
+
+type ImageEditorApi = {
+    patchImage: (
+        image: DbEnrichedImage,
+        patch: Partial<DbEnrichedImage>
+    ) => void
+    deleteImage: (image: DbEnrichedImage) => void
+    getImages: () => void
+}
+
+function AltTextEditor({
+    image,
+    text,
+    patchImage,
+}: {
+    image: DbEnrichedImage
+    text: string
+    patchImage: ImageEditorApi["patchImage"]
+}) {
+    const [value, setValue] = useState(text)
+
+    const handleBlur = useCallback(
+        (e: React.FocusEvent<HTMLTextAreaElement>) => {
+            const trimmed = e.target.value.trim()
+            setValue(trimmed)
+            if (trimmed !== text) {
+                patchImage(image, { defaultAlt: trimmed })
+            }
+        },
+        [image, text, patchImage]
+    )
+
+    return (
+        <TextArea
+            value={value}
+            onChange={(e) => setValue(e.target.value)}
+            onBlur={handleBlur}
+        />
+    )
+}
+
+function createColumns({
+    api,
+}: {
+    api: ImageEditorApi
+}): ColumnsType<DbEnrichedImage> {
+    return [
+        {
+            title: "Preview",
+            dataIndex: "cloudflareId",
+            width: 100,
+            key: "cloudflareId",
+            render: (cloudflareId, { originalWidth }) => {
+                const srcFor = (w: number) =>
+                    `https://imagedelivery.net/qLq-8BTgXU8yG0N6HnOy8g/${encodeURIComponent(cloudflareId)}/w=${w}`
```

Comment:
Should the path be hardcoded here? Doesn't it come from an env var elsewhere?

Reply:
🙈